karabo_data/geometry2.py

```diff
@@ -205,6 +217,139 @@ def inspect(self, frontview=True):
+        """
+        geom_dict = load_crystfel_geometry(filename)
+        modules = []
+        for p in range(cls.n_modules):
+            tiles = []
+            modules.append(tiles)
+            for a in range(cls.n_tiles_per_module):
+                d = geom_dict['panels']['p{}a{}'.format(p, a)]
+                tiles.append(GeometryFragment.from_panel_dict(d))
+        return cls(modules, filename=filename)
+
+    def _get_rigid_groups(self, nquads=4):
+        """Create rigid stings for rigid groups definiton."""
+
+        quads = ','.join(['q{}'.format(q) for q in range(nquads)])
+        modules = ','.join(['p{}'.format(p) for p in range(self.n_modules)])
+
+        prod = product(range(self.n_modules), range(self.n_tiles_per_module))
+        rigid_group = ['p{}a{}'.format(p, a) for (p, a) in prod]
+        rigid_string = '\n'
+
+        for nn, rigid_group_q in enumerate(np.array_split(rigid_group, nquads)):
+            rigid_string += 'rigid_group_q{} = {}\n'.format(nn, ','.join(rigid_group_q))
+        rigid_string += '\n'
+        for nn, rigid_group_p in enumerate(np.array_split(rigid_group, self.n_modules)):
+            rigid_string += 'rigid_group_p{} = {}\n'.format(nn, ','.join(rigid_group_p))
+
+        rigid_string += '\n'
+
+        rigid_string += 'rigid_group_collection_quadrants = {}\n'.format(quads)
+        rigid_string += 'rigid_group_collection_asics = {}\n\n'.format(modules)
+        return rigid_string
+
+    def write_crystfel_geom(self, filename, *,
+                            data_path='/entry_1/instrument_1/detector_1/data',
+                            mask_path=None, dims=('frame', 'modno', 'ss', 'fs'),
+                            adu_per_ev=None, clen=None, photon_energy=None):
+        """Write this geometry to a CrystFEL format (.geom) geometry file.
+        Parameters
+        ----------
+        filename : str
+        filename of the geometry file
+
+        data_path  : str, default : /entry_1/instrument_1/detector_1/data
+            path to the group that contains the data array in the hdf5 file
+        mask_path : str
+            path to the group that contains the mask array in the hdf5 file
+
+        dims : collection, default : ('frame', 'modno', 'ss', 'fs')
+            Dimensions of the data. Extra dimensios, except for details are
+            added by thier index. The dimensions should contain frame, modno,
+            ss, fs.
+
+        adu_per_ev : float, default : 0.0075
+            ADU (analog digital units) per electron volt for the considered
+            detector.
+
+        clen : float, default : 0.119
+            Distance between sample and detector in meters
+
+        photon_energy : int, default : 9800
+            Beam wave length in eV
+        """
+        from . import __version__
+
+        if adu_per_ev is None:
+            warnings.warn('adu_per_ev must be set; Setting to default value')
+            adu_per_ev_str = '; adu_per_eV = 0.0075'
```

Comment:
If we're having no default, I'd not even write a number in the commented line. Just put `; adu_per_eV = XXX` or `; adu_per_eV = SET ME` or something. Otherwise, people will just uncomment it with the value that's already there. Then we can get rid of the warnings as well.

Reply:
Ok!